src/newgrf_roadstop.cpp

```diff
@@ -0,0 +1,268 @@
+
+#include "stdafx.h"
+#include "debug.h"
+#include "station_base.h"
+#include "newgrf_roadstop.h"
+#include "newgrf_class_func.h"
+#include "newgrf_cargo.h"
+#include "newgrf_roadtype.h"
+#include "gfx_type.h"
+#include "company_func.h"
+#include "road.h"
+#include "window_type.h"
+
+#include "safeguards.h"
+
+template <typename Tspec, typename Tid, Tid Tmax>
+void NewGRFClass<Tspec, Tid, Tmax>::InsertDefaults() {
+	// TODO: Give the added RoadStopSpec's a `stop_type`. For now GRF added roadstopspec's are not shown in this class, if new classes are not available.
+	//Default stop class.
+	classes[0].global_id = 'DBUS';
+	classes[0].name = STR_STATION_CLASS_DFLT;
+	classes[0].Insert(nullptr);
+	classes[0].Insert(nullptr);
+}
+
+template <typename Tspec, typename Tid, Tid Tmax>
+bool NewGRFClass<Tspec, Tid, Tmax>::IsUIAvailable(uint index) const {
+	return true;
+}
+
+INSTANTIATE_NEWGRF_CLASS_METHODS(RoadStopClass, RoadStopSpec, RoadStopClassID, ROADSTOP_CLASS_MAX)
+
+static const uint NUM_ROADSTOPSPECS_PER_ROADSTOP = 255; ///< Maximum number of parts per station.
+
+/* virtual */ const SpriteGroup *RoadStopResolverObject::ResolveReal(const RealSpriteGroup *group) const {
```

Comment:
Please don't use the `/* virtual */` annotations in new code, use the proper `override` keyword in the function signature.